### PR TITLE
title:Node.js User Authentication Guide

### PR DESCRIPTION
**Pull request by uma-victor1**
description:This is a blog about you can authenticate users in Nodejs

#### Please note that in case of below checklist not updated accordingly, the maintainers have the right to immediately close your pull request.

- [x] You created a branch from `dev` branch.
- [x] You PR is raised on `dev` branch and not on`master`.
- [x] Blog follows [Blog Guidelines](https://github.com/LoginRadius/engineering-portal/blob/master/GUIDELINES.md).
- [x] You have read the [Contribution Guidelines](https://github.com/LoginRadius/engineering-portal/blob/master/CONTRIBUTING.md) before creating this PR.
- [x] Checked for any duplicate blog entry.
- [x] Checked for all grammatical errors in the blog.
- [x] You have not published the blog content anywhere else on the internet on platforms like medium, dev.to etc.
- [x] The content is not copy-pasted from anywhere else on the Internet.

### CODE REVIEW

File: content/guest-post/node.js-user-authentication-guide/Node_User_Role_auth.md [hunk 1]
```diff
@@ -0,0 +1,1224 @@
...
+
+The server will use the `router` middleware function if there is a request to `/api/auth`.
+
+### Test the Register Route
+
+You'll use [Postman](https://www.postman.com/downloads/) to test all the routes.
+
+Open up Postman to make a `POST` request to `http://localhost:5000/api/auth/register` and pass the username and password to the body:
+
+![Register Route](https://paper-attachments.dropbox.com/s_1FDB76B1B7350D0381EC73781AAAABB04FC20EA7CD90D74041F3D461F538847D_1639605986714_registerRoute.jpg)
```
Comment (raghunath-r-a) [inline content/guest-post/node.js-user-authentication-guide/Node_User_Role_auth.md:259]: @uma-victor1 Throughout the blog, please place the image file in the same folder as the blog content and replace the Dropbox link

File: content/guest-post/node.js-user-authentication-guide/Node_User_Role_auth.md [hunk 2]
```diff
@@ -0,0 +1,1224 @@
...
+
+After creating the `GET` request, add a logout button to the admin's route and user's route:
+
+```html
+...
+<ul></ul>
+<button class="logout"><a href="/logout">Log Out</a></button>
+...
+```
+
```
Comment (raghunath-r-a) [inline content/guest-post/node.js-user-authentication-guide/Node_User_Role_auth.md:1217]: @uma-victor1 Here, please add a section and explain briefly how the above process can be easily replaced with LoginRadius if the readers choose to. You could refer this resource for a brief understanding: https://www.loginradius.com/docs/developer/tutorial/node-js/

File: content/guest-post/node.js-user-authentication-guide/Node_User_Role_auth.md [hunk 3]
```diff
@@ -0,0 +1,1224 @@
...
+## Conclusion
+
+You've successfully learned how to perform CRUD operations using MongoDB as database, hash passwords, authenticate using JWT, and render Embedded JavaScript (EJS). You can put all these together to build more complex applications.
+
+## Resources
+
+- [GitHub Repo](https://github.com/uma-victor1/role_auth)
```
Comment (raghunath-r-a) [inline content/guest-post/node.js-user-authentication-guide/Node_User_Role_auth.md:1224]: @uma-victor1 Please add your sample code in the following repo by raising a separate pull request.
​​
​​https://github.com/LoginRadius/engineering-blog-samples

**Review by raghunath-r-a — CHANGES_REQUESTED** [review]

File: content/guest-post/node.js-user-authentication-guide/Node_User_Role_auth.md (hunk 1 above)
Comment (uma-victor1) [reply content/guest-post/node.js-user-authentication-guide/Node_User_Role_auth.md:259]: fixed

File: content/guest-post/node.js-user-authentication-guide/Node_User_Role_auth.md (hunk 2 above)
Comment (uma-victor1) [reply content/guest-post/node.js-user-authentication-guide/Node_User_Role_auth.md:1217]: Done.

File: content/guest-post/node.js-user-authentication-guide/Node_User_Role_auth.md (hunk 3 above)
Comment (uma-victor1) [reply content/guest-post/node.js-user-authentication-guide/Node_User_Role_auth.md:1224]: Done.

**Comment by raghunath-r-a** [comment]
Duplicated by #950

**Pull request closed without merging**